src/main/scala/me/jeffmay/neo4j/client/ws/WSNeo4jClient.scala

```diff
@@ -0,0 +1,208 @@
+import play.api.libs.ws.{WSAuthScheme, WSClient, WSRequest, WSResponse}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future, blocking}
+import scala.util.Try
+
+class WSNeo4jClient(
+  wsClient: WSClient,
+  val config: Neo4jClientConfig,
+  implicit private val scheduler: Scheduler,
+  implicit private val executionContext: ExecutionContext
+) extends Neo4jClient with Proxy {
+  client =>
+
+  import WSNeo4jClient._
+  import WSRequestWithException._
+
+  override def self: Any = (wsClient, config)
+
+  def copy(
+    ws: WSClient = this.wsClient,
+    config: Neo4jClientConfig = this.config,
+    scheduler: Scheduler = this.scheduler,
+    executionContext: ExecutionContext = this.executionContext
+  ) = {
+    new WSNeo4jClient(wsClient, config, scheduler, executionContext)
+  }
+
+  lazy val ws: WSClient = new ProxyWSClient(wsClient) {
+    override def url(url: String): WSRequest = {
+      new TimeoutWSRequest(wsClient.url(url), config.timeout)
+    }
+  }
+
+  override def withAutoInitialize(autoAuthenticate: Boolean): Neo4jClient =
+    this.copy(config = config.copy(autoInitialize = autoAuthenticate))
+
+  override def withCredentials(username: String, password: String): Neo4jClient =
+    this.copy(config = config.copy(credentials = Neo4jBasicAuth(username, password)))
+
+  override def withBaseUrl(baseUrl: String): Neo4jClient =
+    this.copy(config = config.copy(baseUrl = baseUrl))
+
+  override def withStatsIncludedByDefault(includeStatsByDefault: Boolean): Neo4jClient =
+    this.copy(config = config.copy(includeStatsByDefault = includeStatsByDefault))
+
+  override def withTimeout(timeout: FiniteDuration): Neo4jClient =
+    this.copy(config = config.copy(timeout = timeout))
+
+  protected def whenInitialized[T](action: => Future[T]): Future[T] = {
+    if (config.autoInitialize) {
+      WSNeo4jClient.withVerifiedCredentials(withCredentials("neo4j", "neo4j"), config.credentials)(action)
+    }
+    else action
+  }
+
+  protected def http(path: String): WSRequest = {
+    require(!path.isEmpty, "path cannot be empty")
+    require(path.charAt(0) == '/', "path must be an absolute path starting with '/'")
+    import config.credentials.{password, username}
+    ws.url(config.baseUrl + path).withAuth(username, password, WSAuthScheme.BASIC)
+  }
+
+  override def passwordChangeRequired(): Future[Boolean] = {
+    val request = http(s"/user/${config.credentials.username}")
+    whenInitialized {
+      request.getAndCheckStatus(Set(200)).map { resp =>
+        (resp.json \ "password_change_required").asOpt[Boolean] contains true
+      }
+    }
+  }
+
+  override def changePassword(newPassword: String): Future[Unit] = {
+    val request = http(s"/user/${config.credentials.username}/password")
+    request.postAndCheckStatus(Some(Json.obj("password" -> newPassword)), Set(200))
+      .map(_ => ())
+  }
+
+  override def openTxn(statements: Seq[CypherStatement]): Future[OpenedTxnResponse] = {
```

Comment:
I'm not sure I like this API. It accepts a `Seq` of transactions and then you have to dig through the response to find a `Seq` of results. The assumption is that they are in the same order, but then you have to use integer index to relate the two together. This can be a pretty bad user experience because the caller has to code defensively against index out of bounds, etc.

I like that I can accept a batch of statements, but I'm wondering if there should be a more convenient method that calls this and zips the results together so that the whole statement gets bundled into a single result. Maybe I could sum up all the stats, combine the results, etc. I'm not sure how this would work when parsing the results from the queries tho. It certainly would be convenient for when one is passing a single cypher statement.